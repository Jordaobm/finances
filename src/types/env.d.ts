declare module "@env" {
  export const WEBCLIENTID: string;
}

declare module "*.svg";
declare module "*.png";
declare module "*.jpg";
