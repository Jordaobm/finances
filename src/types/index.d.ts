declare module "*.png";
declare module "*.svg";
declare module "react-native-radial-gradient";
