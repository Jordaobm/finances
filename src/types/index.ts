export interface IconProps {
  color: string;
}

export interface HeaderProps {
  color: "white" | "black";
  onlySettings?: boolean;
}

export interface Operation {
  category: {
    color: string;
  };
  card: {
    name: string;
  };
  value: number;
  date: Date;
  name: string;
}

export interface CardOperationProps {
  operations: Operation[];
}

export interface NavigationProps {
  activeRoute: string;
}
