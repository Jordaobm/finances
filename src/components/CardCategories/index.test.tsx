import { NavigationContainer } from "@react-navigation/native";
import React from "react";
import renderer, { act } from "react-test-renderer";
import { CardCategories } from ".";
import { Card, Category } from "../../types";

jest.mock("react-native-toast-message", () => ({
  show: jest.fn(),
  hide: jest.fn(),
}));

jest.mock("../../context/UpdateDataContext", () => ({
  useUpdateDataContext() {
    const operations = [
      {
        card: {
          colorBackground: "#282b46",
          colorBackgroundNumber: 47,
          colorText: "rgba(270,270,270,1)",
          colorTextNumber: 270,
          currentValue: 90.6,
          id: "1",
          institutionName: "Carteira",
          name: "Carteira",
        },
        category: {
          accumuledValue: 500,
          color: "rgba(197, 48, 48, 1  )",
          id: "1635000545505",
          name: "Dinheiro da vovó",
          number: 191.3680076599121,
        },
        date: "23/10/2021",
        for: {},
        id: "1635000577024",
        name: "Dinheiro da vovó",
        origin: {},
        type: "OUTPUT",
        value: 500,
      },
    ];

    let updateCategory: Category = {} as Category;

    function setUpdateCategory(category: Category) {
      updateCategory = category;
    }

    return {
      operations,
      setUpdateCategory,
    };
  },
}));

{
}

test("Component CardCategories", async () => {
  const categories: Category[] = [
    {
      accumuledValue: 500,
      color: "rgba(197, 48, 48, 1  )",
      id: "1635000545505",
      name: "Dinheiro da vovó",
      number: 191.3680076599121,
    },
  ];

  const tree = renderer
    .create(
      <NavigationContainer>
        <CardCategories categories={categories} />
      </NavigationContainer>
    )
    .toJSON();
  await act(async () => {
    expect(tree).toMatchSnapshot();
  });
});
