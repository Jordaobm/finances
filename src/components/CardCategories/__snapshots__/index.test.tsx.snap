// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component CardCategories 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        color="rgba(197, 48, 48, 1  )"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(197, 48, 48, 1  )",
              "borderRadius": 10,
              "height": 10,
              "marginRight": 16,
              "width": 10,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#595959",
              "fontFamily": "Roboto-Medium",
              "fontSize": 18,
            },
          ]
        }
      >
        Dinheiro da vovó
      </Text>
    </View>
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        color="rgba(197, 48, 48, 1  )"
        style={
          Array [
            Object {
              "color": "rgba(197, 48, 48, 1  )",
              "fontFamily": "Roboto-Medium",
              "fontSize": 18,
              "left": -4,
              "position": "relative",
            },
          ]
        }
      >
         R$500,00
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#c6c6c6",
              "fontFamily": "Roboto-Regular",
              "fontSize": 12,
            },
          ]
        }
      >
        última adição em
         
        23/10/2021
      </Text>
    </View>
  </View>
</View>
`;
